tests/integration_tests/functional/test_feat_parity.py

```diff
@@ -0,0 +1,407 @@
+        #     (1 << 14) | # IBRS
+        #     (1 << 15) | # STIBP
+        #     (1 << 24) # SSBD
+        # )
+    ]
+
+    must_be_set_intel_only = [
+        # Security related
+        (0x7, 0x0, "edx",
+            (1 << 10) | # MD_CLEAR
+            (1 << 29) # IA32_ARCH_CAPABILITIES
+        )
+    ]
+
+    must_be_set_amd_only = [
+        # Security related
+        (0x80000008, 0x0, "ebx",
+            (1 << 18) | # IbrsPreferred
+            (1 << 19) # IbrsProvidesSameModeProtection
+        )
+    ]
+
+    must_be_unset_common = [
+        # Security related
+        (0x7, 0x0, "edx",
+            (1 << 28) # L1D_FLUSH
+        )
+    ]
+
+    must_be_unset_intel_only = [
+        # Security related
+        (0x80000008, 0x0, "ebx",
+            (1 << 18) | # IbrsPreferred
+            (1 << 19) # IbrsProvidesSameModeProtection
+        )
+    ]
+
+    must_be_unset_amd_only = [
+        # Security related
+        (0x7, 0x0, "edx",
+            (1 << 10) | # MD_CLEAR
+            (1 << 29) # IA32_ARCH_CAPABILITIES
+        )
+    ]
+    # fmt: on
+
+    vendor = cpuid_utils.get_cpu_vendor()
+    if vendor == cpuid_utils.CpuVendor.INTEL:
+        must_be_set = must_be_set_common + must_be_set_intel_only
+        must_be_unset = must_be_unset_common + must_be_unset_intel_only
+    elif vendor == cpuid_utils.CpuVendor.AMD:
+        must_be_set = must_be_set_common + must_be_set_amd_only
+        must_be_unset = must_be_unset_common + must_be_unset_amd_only
+
+    _test_cpuid_feat_flags_matrix(
+        bin_cloner_path, network_config, cpu_template, must_be_set, must_be_unset
+    )
+
+
+def check_arch_cap_msr(vm, cpu_template):
+    """
+    Checks that IA32_ARCH_CAPABILITIES MSR has the expected value.
+    """
+    kernel_ver = get_guest_kernel_ver(vm)
+
+    ssh_conn = net_tools.SSHConnection(vm.ssh_config)
+    rdmsr_cmd = "rdmsr 0x10a"
+    _, stdout, stderr = ssh_conn.execute_command(rdmsr_cmd)
+
+    if cpu_template == "T2CL":
+        assert stderr.read() == "", f"{kernel_ver=}"
+        actual = int(stdout.read().strip(), 16)
+        expected = 0xEB
+        assert (
+            actual == expected
+        ), f"{cpu_template}: {kernel_ver=}, {actual=:#x}, {expected=:#x}"
+    elif cpu_template == "T2A":
+        # IA32_ARCH_CAPABILITIES shall not be available
+        assert stderr.read() != "", f"{cpu_template}: {kernel_ver=}"
+
+
+def _test_msr_arch_cap(context):
+    vm_builder = context.custom["builder"]
+    root_disk = context.disk.copy()
+    cpu_template = context.custom["cpu_template"]
+
+    vm_instance = vm_builder.build(
+        kernel=context.kernel,
+        disks=[root_disk],
+        ssh_key=context.disk.ssh_key(),
+        config=context.microvm,
+        cpu_template=cpu_template,
+    )
+    vm = vm_instance.vm
+    vm.start()
+
+    check_arch_cap_msr(vm, cpu_template)
+
+
+@pytest.mark.skipif(
+    platform.machine() != "x86_64", reason="CPU templates are only available on x86_64."
+)
```

Comment:
Maybe this can be simplified with:
```suggestion
X86_64 = platform.machine() == "X86_64"
@pytest.mark.skipif(
    not X86_64, reason="CPU templates are only available on x86_64."
)
```

Reply:
done